Convert node side Generator

RocksSavage opened the pull request:
<!--
If this PR is associated with an issue be sure to reference it
by including "refs #18436" or "closes #18436 " (e.g., #closes #1234).

If this PR implements an enhancement that does not have an existing issue, please add the following:
-->

## Reason
#closes #18436


<!--
If this PR implements a bug fix, please create an issue for the bug and reference the issue.
-->

RocksSavage committed:
inital commit; created a generator class and added a test case .i file
refs #18436

RocksSavage committed:
Renamed files and created a test out of the original discussion that prompted this ticket
refs #18436

RocksSavage committed:
correct minor mistake in ConvertNodeSet... file and make a testy file for playing with
refs #18436

RocksSavage committed:
Complete two working tests for sideset nodeset converter
refs #18436

RocksSavage committed:
Created documentation for ConvertSideSetNodeSet
ref #18436

Modified documentation text refs #18436

RocksSavage committed:
Small modifications to mooseMesh, cleanup an artifact. ref #18436

applies style changes. Thanks Moosebuild ref #18436

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1063821/) on c5af950 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/21196/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/21196/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 73dd22b5a5f11b7534f534e2229fc2a9e57f6c13`

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1064354/) on 38b5450 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/21196/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/21196/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 73dd22b5a5f11b7534f534e2229fc2a9e57f6c13`

RocksSavage committed:
Revert changes lost (through rebase?), and corrected typos ref #18436

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1064665/) on a6326c8 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/21196/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/21196/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format f14a86060df928685f8289e0f57ad70b24d4c69b`

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1064747/) on 2606e06 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/21196/site)

This comment will be updated on new commits.

GiudGiud commented:
Test failure is unrelated

any takers for reviewing @cticenhour @lindsayad @hugary1995

hugary1995 commented on framework/src/meshgenerators/ConvertNodeSetSideSetGenerator.C:
```diff
@@ -0,0 +1,54 @@
+      "Mesh generator which constructs side sets from node sets, and vise versa");
+  params.addRequiredParam<MeshGeneratorName>("input",
+                                             "Input mesh the operation will be applied to");
+  params.addRequiredParam<bool>("convert_node_list_from_side_list",
+                                "Flag signaling if nodes should be constructed from sides");
+  params.addRequiredParam<bool>("convert_side_list_from_node_list",
+                                "Flag signaling if sides should be constructed from nodes");
```
I am not a fan of using booleans to combine capabilities. The MOOSE-way of doing it is to have two separate classes. A `NodeSetToSideSetGenerator` and a `SideSetToNodeSetGenerator`. They could be deriving from the same parent class to reduce code duplication.

GiudGiud replied:
On one hand selecting the nodesets / sidesets would be nice and make it more targeted.

On the other hand the other mesh generators that @RocksSavage are more important for our applications so I dont want to spend too long on these

So I'd say address the other comments and we'll advise later when we have good progress on the others

hugary1995 reviewed (changes requested):
Some suggestions for starters.

hugary1995 commented:
Besides my comments above -- this is a really nice contribution!

RocksSavage committed:
Creates two classes out of the erstwhile ConvertNodeSetSideSet class, updates tests and docs related to them ref #21196

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1072051/) on 58855ed wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/21196/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/21196/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format fd49fbec33a53f8de346cffc58499a52da6c719f`

cticenhour commented:
@GiudGiud I will give this another formal review when I return next week. @hugary1995 do you have any more comments?

GiudGiud commented:
I ll clean it up today.
Please dont spend time reviewing before it s clean

GiudGiud committed:
Clean up of sideset <-> nodeset conversion generators
rename to match other generators
fix documentation that confused nodesets and sidesets
fix class descriptions
clean up test inputs to make them minimal
clean up tests specification and requirements
closes #18436

GiudGiud commented:
Ok let's see how this does with the test suite.

hugary1995 commented:
I'm being extra picky: I'd prefer the name `SidesetFromNodesetGenerator` over `NodesetToSidesetGenerator`. The current names probably give people a false impression that the source side set and node set are _removed_ as a consequence.

GiudGiud commented:
but the docs say they are not :O

hugary1995 commented:
Fair enough

GiudGiud commented:
The argument which could defeat me would be that all our generators are currently SideSetsFrom and not SideSetsTo

do you want to use this argument

GiudGiud commented:
ok made the change.

Would love a thumbs up on this PR once it passes the distributed tests

GiudGiud commented:
Thanks for the reviews!
